tests/Support/SupportCollectionTest.php

```diff
@@ -1421,6 +1421,40 @@ public function testWithMultipleModeValues()
         $collection = new Collection([1, 2, 2, 1]);
         $this->assertEquals([1, 2], $collection->mode());
     }
+
+    public function testSingle()
+    {
+        $collection = new Collection(['foo', 'bar', 'baz']);
+
+        $result = $collection->single(function ($item) {
+            return $item === 'bar';
+        });
+        $this->assertEquals('bar', $result);
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     */
+    public function testSingleThrowsAnErrorWhenMoreThanOneMatchingValueIsFound()
+    {
+        $collection = new Collection(['foo', 'foo', 'bar']);
+
+        $result = $collection->single(function ($item) {
+            return $item === 'foo';
+        });
+    }
+
+    /**
+     * @expectedException InvalidArgumentException
+     */
+    public function testSingleThrowsAnErrorWhenNoMatcingValueIsFound()
+    {
+        $collection = new Collection(['foo', 'bar']);
+
+        $result = $collection->single(function ($item) {
+            return $item === 'baz';
+        });
```

Comment:
In this case, `baz` is not in the original collection so an exception would be thrown. $result would not equal `baz`.

Thanks for the feedback so far though, implementing it currently